music21/midi/translate.py

```diff
@@ -1833,10 +1867,17 @@ def channelInstrumentData(s: stream.Stream,
     substreamList = []
     if s.hasPartLikeStreams():
         for obj in s.getElementsByClass('Stream'):
-            substreamList.append(obj)
+            # _prepareStreamForMidi() supplies defaults for these
+            if obj.getElementsByClass(('MetronomeMark', 'TimeSignature')):
+                # Conductor track: don't consume a channel
+                continue
```

Comment:
Maybe make this an optional flag in the argument OR update the documentation, so that no one uses this on another stream besides one immediately being parsed.

Reply:
I've made this check more agnostic so that it's just `if not obj.notesAndRests`; then it's safe to assume it shouldn't consume a channel--whether it was run through _preapreStreamForMidi() or not. What do you think?

Reply:
Haha nope that caused a bunch of test failures. It's in the commit history if you want to see it.

Reply:
My instinct I think was good; I just needed to check `not obj.flat.notesAndRests` and now it's agnostic to any stream input.